test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -68,11 +70,32 @@
       app {:jruby-puppet
            {:max-active-instances num-jrubies}
            :versioned-code
-           {:code-id-command (script-path "warn_echo_and_error")}}
+           {:code-id-command (script-path "warn_echo_and_error")
+            :code-content-command (script-path "echo")}}
       (let [catalog (testutils/get-catalog)]
         (is (nil? (get catalog "code_id")))
         (is (logged? #"Non-zero exit code returned while running" :error))
-        (is (logged? #"Executed an external process which logged to STDERR: production" :warn)))))))
+        (is (logged? #"Executed an external process which logged to STDERR: production" :warn))))))
+  (testing "code id is not added and 500 is returned if code-content-command is not set"
+    (logging/with-test-logging
+     (bootstrap/with-puppetserver-running
+      app {:jruby-puppet
+           {:max-active-instances num-jrubies}
+           :versioned-code
+           {:code-id-command (script-path "echo")}}
+      (let [response (testutils/http-get "puppet/v3/catalog/localhost?environment=production")]
+        (is (= 500 (:status response)))
+        (is (logged? #"Error 500 on SERVER"))))))
```

Comment:
I can certainly add a test for the other case.
I'm not sure about 409 though, as the description states "This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request." Here the user might not have any control over the configuration on the server to be able to add code-id-command or code-content-command.